bin/tasks/build.js

```diff
@@ -51,6 +63,13 @@ export const buildStorybook = async ctx => {
   } finally {
     logFile.end();
   }
+
+  const buildLog = fs.readFileSync(ctx.buildLogFile, 'utf8');
+  const outputDir = getOutputDir(buildLog);
+  if (outputDir && outputDir !== ctx.sourceDir) {
+    ctx.log.warn(deviatingOutputDir(ctx, outputDir));
+    ctx.sourceDir = outputDir;
+  }
```

Comment:
This process seems a bit brittle (especially when you consider future versions of SB may not have the same log format). What do you think about only doing if it we *don't* find the build in the place we expected?, ie  ala https://github.com/chromaui/chromatic-cli/pull/164)

Reply:
Moved to upload task so we only run this if the sourceDir isn't a valid build.